packages/client/test/unit/SynchronizedGraphQLClient.test.ts

```diff
@@ -0,0 +1,177 @@
+import 'reflect-metadata'
+import { SynchronizedGraphQLClient } from '../../src/utils/SynchronizedGraphQLClient'
+import { mockContext } from '../test-utils/utils'
+
+const POLL_INTERVAL = 50
+const INDEXING_INTERVAL = 100
+const MOCK_QUERY = 'mock-query'
+
+interface IndexState {
+    blockNumber: number
+    queryResult: any
+}
+
+class FakeIndex {
+
+    private states: IndexState[]
+    private blockNumber = 0
+    // eslint-disable-next-line no-undef
+    private timer: NodeJS.Timer | undefined
+
+    constructor(states: IndexState[]) {
+        this.states = states
+    }
+
+    getState(): IndexState {
+        return this.states.find((state) => state.blockNumber >= this.getBlockNumber())!
```

Comment:
Are you sure about the blockNumber logic here? It seems as if in `this.blockNumber = 3` it would be returning a result with `{ blockNumber: 4, ... }` with the below `beforeEach` block?